contracts/NFTs/AftermarketDeviceId.sol

```diff
@@ -16,4 +19,35 @@ contract AftermarketDeviceId is Initializable, NftBaseUpgradeable {
     ) external initializer {
         _baseNftInit(name_, symbol_, baseUri_);
     }
+
+    /// @notice Sets the DIMO Registry address
+    /// @dev Only an admin can set the DIMO Registry address
+    /// @param addr The address to be set
+    function setDimoRegistryAddress(address addr)
+        external
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        require(addr != address(0), "Non zero address");
+        _dimoRegistry = IDimoRegistry(addr);
+    }
+
+    /// @notice Internal function to transfer a token
+    /// @dev Only the token owner can transfer (no approvals)
+    /// @dev The AD must not be paired
+    /// @param from Old owner
+    /// @param to New owner
+    /// @param tokenId Token Id to be transferred
+    function _transfer(
+        address from,
+        address to,
+        uint256 tokenId
+    ) internal override {
+        // Approvals are not accepted for now
+        require(
+            msg.sender == address(_dimoRegistry) || msg.sender == from,
+            "Caller is not the owner"
+        );
+        _dimoRegistry.verifyAftermarketDeviceTransfer(tokenId);
```

Comment:
Does this method on the registry seem overly specific to you? Asking whether—and to what—a vehicle is paired seems like something of general interest, even after we remove the transfer restriction.

I guess it's unclear who should "own" this guard.

Reply:
Humm... for now we just do the check actually in the `mapper` contract storage. But I think we might want to do more checks in the future of even change the state, so kinda makes sense this function be in the `AftermarketDevice`

Reply:
I guess I'm asking whether one might prefer to just have something like `dimoRegistry.getVehiclePairedWithAftermarketDevice(tokenId) == 0`. "What vehicle is this connected to, if any?" seems like a natural on-chain question.

Reply:
Anyway, don't let this hold up getting this out there, I think it's good.